### PR TITLE
Bump Version of Package

### LINKED ISSUE #132: Opt-in feature: bump patch component of version number

### PR DESCRIPTION
**Pull request by fchorney**
It seems typical to bump the package version when adding dependency compat values. I have set this to false by default but if we want it true by default I can change it.

If the package is >= 1.0 then the minor value will be bumped and the patch will go to 0. If the package is < 1.0 then only the patch value will be bumped.

I had a request to ignore the bumping of version values if a pre-release strings exists in the version number such as `-DEV`. Let me know if this needs to be relaxed or if we want a different rule for it.

Note: This was made against the v3-dev branch. 

closes #132

### CODE REVIEW

File: src/utilities/new_versions.jl [hunk 1]
```diff
@@ -340,21 +342,51 @@ function unsub_from_pr(api::GitLab.GitLabAPI, pr::GitLab.MergeRequest)
...
+function bump_package_version!(project::Dict)
+    version = VersionNumber(project["version"])
+
+    # Only bump the version if prerelease is empty
+    !isempty(version.prerelease) && return nothing
+
+    # Bump minor if version > 1.0, else bump patch
+    if version.major >= 1
+        version = VersionNumber(
+            version.major, version.minor + 1, 0, version.prerelease, version.build
```
Comment (mattBrzezinski) [inline src/utilities/new_versions.jl:374]: I don't think prerelease or build are used in compat entries, so they're not needed?
I feel like keeping the same build number especially is wrong, as that would not really exist?

Comment (fchorney) [reply src/utilities/new_versions.jl:374]: This updates the package version, not the compat version. That being said, maybe this shouldn't be in the same function that updates the compat version

**Comment by fchorney** [comment]
bors try

**Comment by DilumAluthge** [comment]
> I have set this to false by default but if we want it true

Yeah I think `false` by default is good.

Would it make sense to just always bump the patch version? I don't think a compat bump needs a minor version bump in a post 1.0 packages, does it?

**Comment by fchorney** [comment]
> Would it make sense to just always bump the patch version? I don't think a compat bump needs a minor version bump in a post 1.0 packages, does it?

According to [ColPrac ](https://github.com/SciML/ColPrac) it seems that it should. It has this section about changing compat entries:

### Changing dependency compatibility

*   Generally changing dependency compatibility should be a non-breaking feature.
    *   i.e. pre-1.0 change patch version number, post-1.0 change the minor version number.
    *   For instance, adding or removing compatibility with a particular __version__ of a current dependency, which may or may not require internal code changes.
    *   This also applies when adding or removing packages as dependencies.
    *   The new feature in question is the ability to use with a different set of packages.
*   Changing a dependency to resolve a bug is a bug-fix.
    *   i.e. pre/post-1.0 change patch version number.
    *   For instance, if a bug in a downstream dependency is causing a problem in your package restricting compat to not allow that version would be a bug-fix.
*   Changing compatibility with dependencies **may** be a breaking release, if it breaks the user-facing interface.
    That is to say if the dependency’s API leaks into your API.
    There are three ways that this can happen:
    *   Reexporting a function that has changed.
    *   Returning an object of a type that’s behaviour has changed.
    *   Subtyping an object that has changed.

**Comment by fchorney** [comment]
bors try

**Comment by fchorney** [comment]
bors try

**Comment by DilumAluthge** [comment]
> I had a request to ignore the bumping of version values if a pre-release strings exists in the version number such as `-DEV`.

I think this rule sounds good.

**Comment by fchorney** [comment]
Any other comments or does this seem good to go?

**Review by mattBrzezinski — APPROVED** [review]

**Pull request merged**